src/System.Private.CoreLib/shared/System/Threading/ThreadPool.cs

```diff
@@ -668,6 +679,13 @@ internal static bool Dispatch()
                 //
                 needAnotherThread = false;
             }
+#endif
+            catch (Exception e)
+            {
+                // Work items should not allow exceptions to escape.  For example, Task catches and stores any exceptions.
+                Environment.FailFast("Unhandled exception in ThreadPool dispatch loop", e);
+                return true; // Will never actually be executed because Environment.FailFast doesn't return
```

Comment:
Don't the hosting interfaces allow you to control what happens when an exception goes unhanded on a thread? Why are we changing this?

Reply:
I started with the diff between CoreRT and CoreCLR and tried to minimize it. This part comes from the CoreRT code. You are right we probably shouldn't catch the exceptions, but I will need to verify what would happen in CoreRT in that case. I'll either remove it or wrap it in `#if CORERT`.

Reply:
The CoreRT implementation of ThreadPool.Windows doesn't have any additional handling of exception between the native thread pool Win32 calls and the `Dispatch`, so it definitely cannot be removed altogether. I'll refactor it in CoreRT and remove it from shared code.